### PR TITLE
Get help existing shortcuts

### PR DESCRIPTION
**Pull request by sgober**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:

This PR refactors the existing context portal functionality to use the shortcut driven completion component.  It should work exactly the same as it previously did, so if it doesn't there was a problem with the refactor.  Test the @condition context portal as well as the #date one.

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/context/ContextCalendar.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
```
Comment (danlee1025) [inline src/context/ContextCalendar.jsx]: missing semicolon

File: src/context/ContextCalendar.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            <Calendar
+                showDateInput={false}
+                onSelect={this.handleDateSelect}
+                ref={input => input && setTimeout(() => {input.focus()}, 100)}
+            />
+        );
+    }
+}
+
+ContextCalendar.proptypes = {
```
Comment (danlee1025) [inline src/context/ContextCalendar.jsx]: `proptypes` should be `propTypes`

File: src/context/ContextListOptions.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
```
Comment (danlee1025) [inline src/context/ContextListOptions.jsx]: missing semicolon

**Comment by danlee1025** [comment]
Refactor looks good.

**Comment by Dtphelan1** [comment]
[DP1] Any reason we don't have the ContextMenus track their own selectedIndicies? Feels odd that we're passing a bunch of props to components that are then never used. If this isn't a priority right now can we at least make a JIRA?

[DP2] Looks like we're still passing the onChange function even though we combined the functionality of changing the state and setting it into the onSelected function.

[DP3] onClose still uses the old onChange idiom; it should be updated to remove the onChange call with the result of the onSelected function. 

[DP4]  I'm getting two errors in the console as a product of the fixed props: 
image
image

**Comment by sgober** [comment]
Addressed DPs 2-4.  For DP1, @mgramigna and I discussed this, the problem is that the selected index is used in the onKeyDown function which is needed in the parent component `ContextPortal.jsx`.  We did only discuss it briefly, but maybe he has more he can add to this

**Comment by mgramigna** [comment]
I talked to Dylan about DP1. After discussing it, we think that all of that `selectedIndex` logic, even including the `onKeyDown` stuff, can be moved outside of the `ContextPoral` component itself for now. The calendar never actually uses this logic, so it's essentially specific to the list options case for the Context Portal

Worth noting for the future: we may want to re-use some of this `onKeyDown` stuff when we implement the get help completion component, so it's possible that we can cleverly re-use some logic between `ContextListOptions` and our future get help menu. Nothing needs to be done in this PR to address this, just something to keep in mind when we actually implement that component

**Comment by jafeltra** [comment]
I'm going to close this PR _temporarily_, as the subtask it is addressing is not in our current sprint. When the subtask is in the sprint, this PR should be reopened and all comments should be addressed then.

Subtask: https://jira.mitre.org/browse/ASCODCP-1948

**Comment by jafeltra** [comment]
NOTE: The branch for this `get-help-existing-shortcuts` should not be deleted, as this will be reopened addressed eventually.

**Pull request closed without merging**